### ISSUE TITLE
[FEATURE] Support range bounds for categorical axis

### ISSUE THREAD

**Issue by droumis** [bokeh/bokeh#13830]
### Problem description

We can set the navigable bounds of a continuous axis, e.g. `p.x_range.bounds = (10, 50)` but not a categorical axis, e.g. `p.y_range.bounds = ("b", "f")`. If the categorical axis ticks are unique, maybe they can be supported in this way?

![image](https://github.com/bokeh/bokeh/assets/6613202/ca4a75c1-2fbd-4bcc-bf9b-c4353a07bfaa)

### Feature description

```python
from bokeh.io import output_notebook
from bokeh.plotting import figure, show
output_notebook()

factors = ["a", "b", "c", "d", "e", "f", "g", "h"]
x = [50, 40, 65, 10, 25, 37, 80, 60]

p = figure(y_range=factors)

p.scatter(x, factors, size=15, fill_color="orange", line_color="green", line_width=3)

p.x_range.bounds = (10, 50)
p.y_range.bounds = ("b", "f")

show(p)
```

### Potential alternatives

Maybe we could define a mapping between categorical and continuous ticks and then use that instead to set the bounds

`factors = [("a", 1), ("b", 2), ...]`

**Comment by bryevdv** [comment]
Is this just about stopping excessive zoom out? The issue with providing "bounds" is that categorical factors do not have any intrinsic order so "start" and "end" do not make sense. Maybe there is another way to suppress excess zoom though, if that is the concern.

**Comment by droumis** [comment]
> Is this just about stopping excessive zoom out? 

Yes, the particular use-case is about limiting the navigable extents
